pkg/generators/git.go

```diff
@@ -1,27 +1,95 @@
 package generators
 
 import (
+	"context"
 	"fmt"
 	argoprojiov1alpha1 "github.com/argoproj-labs/applicationset/api/v1alpha1"
+	"github.com/argoproj-labs/applicationset/pkg/services"
+	"github.com/argoproj-labs/applicationset/pkg/utils"
 	argov1alpha1 "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	"path"
 )
 
 var _ Generator = (*GitGenerator)(nil)
 
 type GitGenerator struct {
+	repos services.Apps
 }
 
-func NewGitGenerator() Generator {
-	g := &GitGenerator{}
+func NewGitGenerator(repos services.Apps) Generator {
+	g := &GitGenerator{
+		repos: repos,
+	}
 	return g
 }
 
-func (g *GitGenerator) GenerateApplications(
-	appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator,
-	appSet *argoprojiov1alpha1.ApplicationSet) ([]argov1alpha1.Application, error) {
-	if appSet == nil {
+func (g *GitGenerator) GenerateApplications(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator, appSet *argoprojiov1alpha1.ApplicationSet) ([]argov1alpha1.Application, error) {
+	if appSetGenerator == nil || appSet == nil {
 		return nil, fmt.Errorf("ApplicationSet is empty")
 	}
 
-	return nil, nil
+	if appSetGenerator.Git == nil {
+		return nil, fmt.Errorf("git variable empty")
+	}
+
+	allApps, err := g.repos.GetApps(context.TODO(), appSetGenerator.Git.RepoURL, appSetGenerator.Git.Revision)
+	if err != nil {
+		return nil, err
+	}
+
+	requestedApps := g.filter(appSetGenerator.Git.Directories, allApps)
+
+	res := g.generateApplications(requestedApps, appSet)
```

Comment:
I can't believe this is such little code. :) I guess a lot of the work must have already been done in argo? Really cool.